koa_crud/src/models/promo.ts

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable no-shadow */
+import { Schema, model, Document } from 'mongoose';
+import mongooseLeanVirtuals from 'mongoose-lean-virtuals';
+
+export enum Template {
+  Deposit = 'DEPOSIT',
+  SignUp = 'SIGN_UP',
+}
+
+export enum Status {
+  Draft = 'DRAFT',
+  Active = 'ACTIVE',
+  Inactve = 'INACTIVE',
+}
+
+export type MemberFields = {
+  realName: string;
+  email: string;
+  bankAccount: number;
+};
+
+export type PromoDocument = Document & {
+  name: string;
+  template: Template;
+  title: string;
+  description: string;
+  submitted: boolean;
+  enabled: boolean;
+  status: Status;
+  minimumBalance: number | null;
+  requireMemberFields: [MemberFields];
```

Comment:
Base on our use case it is awkward to check the array of objects.
I think one of the best alternative is to add enum MemberFields with values `EMAIL`, `REAL_NAME` and `BANK_ACCOUNT`.

and the `requiredMemberFields` should only accept this values.